includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -461,7 +461,7 @@ public function process_upe_redirect_payment( $order_id, $intent_id, $save_payme
 			//     See: https://github.com/Automattic/woocommerce-payments/blob/b69532e92a381cf054f515896c56cc365e3904d4/includes/payment-methods/class-upe-payment-gateway.php#L493
 			// }
 
-			$intent->captured = 'yes'; // TODO: this is to re-use the parent logic, maybe re-implement it to not use charges?, see: https://github.com/Automattic/woocommerce-payments/blob/b69532e92a381cf054f515896c56cc365e3904d4/includes/class-wc-payment-gateway-wcpay.php#L1125
+			$intent->captured = end( $intent->charges->data )->captured; // TODO: this is to re-use the parent logic, maybe re-implement it to not use charges?, see: https://github.com/Automattic/woocommerce-payments/blob/b69532e92a381cf054f515896c56cc365e3904d4/includes/class-wc-payment-gateway-wcpay.php#L1125
```

Comment:
Will create an issue for this TODO. This refactor can be addressed in a separate PR as we would have to make a few changes in the _legacy_ checkout and rewrite part of the implementation in the abstract class.

Reply:
Created #1790